storm-client/src/jvm/org/apache/storm/security/auth/kerberos/AutoTGT.java

```diff
@@ -33,18 +33,21 @@
 import javax.security.auth.login.LoginContext;
 import javax.xml.bind.DatatypeConverter;
 
+import org.apache.storm.Config;
 import org.apache.storm.metric.api.IMetric;
+import org.apache.storm.metric.api.IMetricsRegistrant;
 import org.apache.storm.security.auth.AuthUtils;
 import org.apache.storm.security.auth.IAutoCredentials;
 import org.apache.storm.security.auth.ICredentialsRenewer;
 
+import org.apache.storm.task.TopologyContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Automatically take a user's TGT, and push it, and renew it in Nimbus.
  */
-public class AutoTGT implements IAutoCredentials, ICredentialsRenewer, IMetric {
+public class AutoTGT implements IAutoCredentials, ICredentialsRenewer, IMetric, IMetricsRegistrant {
```

Comment:
We can have individual metric instance and get rid of needs for implementing `IMetric`. It should be removed in Metrics V2 anyway.